src/gc/gc.cpp

```diff
@@ -5950,6 +5954,12 @@ void gc_heap::fix_allocation_context (alloc_context* acontext, BOOL for_gc_p,
     {
         // We need to update the alloc_bytes to reflect the portion that we have not used  
         acontext->alloc_bytes -= (acontext->alloc_limit - acontext->alloc_ptr);  
+
+        // we attribute the unspent bytes to the SOH here.
+        // the loh/soh difference really matters only when we hold corresponding locks
+        // here we have exclusive access to the heap, so we can use either counter.
```

Comment:
this comment is incorrect - we are updating the total_alloc_bytes_soh here because this is strictly only for SOH - LOH does not have the per thread alloc context.